plugins/source/azure/resources/services/network/virtual_network_gateway_connections.go

```diff
@@ -0,0 +1,39 @@
+package network
+
+import (
+	"context"
+
+	"github.com/Azure/azure-sdk-for-go/sdk/resourcemanager/network/armnetwork/v2"
+	"github.com/cloudquery/cloudquery/plugins/source/azure/client"
+	"github.com/cloudquery/plugin-sdk/schema"
+	"github.com/cloudquery/plugin-sdk/transformers"
+)
+
+func VirtualNetworkGatewayConnections() *schema.Table {
+	return &schema.Table{
+		Name:        "azure_network_virtual_network_gateway_connections",
+		Resolver:    fetchVirtualNetworkGatewayConnections,
+		Description: "https://learn.microsoft.com/en-us/rest/api/network-gateway/virtual-network-gateway-connections/list?tabs=HTTP#virtualnetworkgatewayconnection",
+		Multiplex:   client.SubscriptionResourceGroupMultiplexRegisteredNamespace("azure_network_virtual_network_gateway_connections", client.Namespacemicrosoft_network),
+		Transform:   transformers.TransformWithStruct(&armnetwork.VirtualNetworkGatewayConnection{}, transformers.WithPrimaryKeys("ID")),
+		Columns:     schema.ColumnList{client.SubscriptionID},
+	}
+}
+
+func fetchVirtualNetworkGatewayConnections(ctx context.Context, meta schema.ClientMeta, parent *schema.Resource, res chan<- any) error {
+	cl := meta.(*client.Client)
+	svc, err := armnetwork.NewVirtualNetworkGatewayConnectionsClient(cl.SubscriptionId, cl.Creds, cl.Options)
+	if err != nil {
+		return err
+	}
+
+	pager := svc.NewListPager(cl.ResourceGroup, nil)
```

Comment:
See here that we only pass the resource group unlike other resources that pass other parent's fields